lib/chef/provider/load_balancer.rb

```diff
@@ -20,13 +19,16 @@ def new_driver
         @new_driver ||= run_context.chef_metal.driver_for(new_resource.driver)
       end
 
+      def chef_managed_entry_store
+        @chef_managed_entry_store ||= Provisioning.chef_managed_entry_store(new_resource.chef_server)
+      end
+
       action :create do
-        lb_spec = Chef::Provisioning::ChefLoadBalancerSpec.get(new_resource.name) ||
-                  Chef::Provisioning::ChefLoadBalancerSpec.empty(new_resource.name)
+        lb_spec = chef_managed_entry_store.get_or_new(:load_balancer, new_resource.name)
 
         Chef::Log.debug "Creating load balancer: #{new_resource.name}; loaded #{lb_spec.inspect}"
         if new_resource.machines
-          machine_specs = new_resource.machines.map { |machine| get_machine_spec(machine) }
+          machine_specs = new_resource.machines.map { |machine| get_machine_spec!(machine) }
```

Comment:
There are two versions: one that will error out if the machine spec isn't there (requires it to exist), and the other will return nil.